specification/communication/data-plane/Microsoft.CommunicationServicesAdministration/preview/2020-11-01-preview3/phonenumbers.json

```diff
@@ -0,0 +1,1014 @@
+            "description": "Failure",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CapabilityValue": {
+      "description": "",
+      "type": "string",
+      "enum": [
+        "inbound",
+        "outbound",
+        "inbound+outbound",
+        "none"
+      ],
+      "x-ms-enum": {
+        "name": "CapabilityValue",
+        "modelAsString": true
+      }
+    },
+    "Capabilities": {
+      "description": "",
+      "type": "object",
+      "properties": {
+        "sms": {
+          "description": "Available Sms capabilities.",
+          "$ref": "#/definitions/CapabilityValue"
+        },
+        "calling": {
+          "description": "Available Calling capabilities.",
+          "$ref": "#/definitions/CapabilityValue"
+        }
+      }
+    },
+    "AvailableCapabilities": {
+      "description": "",
+      "type": "object",
+      "properties": {
+        "sms": {
+          "description": "Available Sms capabilities.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CapabilityValue"
+          }
+        },
+        "calling": {
+          "description": "Available Calling capabilities.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CapabilityValue"
+          }
+        }
+      }
+    },
+    "PhoneNumberType": {
+      "description": "",
+      "type": "string",
+      "enum": [
+        "tollFree",
+        "geographic"
+      ],
+      "x-ms-enum": {
+        "name": "PhoneNumberType",
+        "modelAsString": true
+      }
+    },
+    "AcquiredPhoneNumber": {
+      "description": "",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The id, which is the phone number in E.164 format without the leading +.",
+          "type": "string"
+        },
+        "phoneNumber": {
+          "description": "The phoneNumber in E.164 format.",
+          "type": "string"
+        },
+        "country": {
+          "description": "The country the phone number belongs to.",
+          "type": "string"
+        },
+        "numberType": {
+          "description": "The type of the phone number.",
+          "$ref": "#/definitions/PhoneNumberType"
+        },
+        "assignmentType": {
+          "description": "",
+          "$ref": "#/definitions/AssignmentType"
+        },
+        "purchaseDate": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "status": {
+          "description": "",
+          "enum": [
+            "activated"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberStatus",
+            "modelAsString": true
+          }
+        },
+        "capabilities": {
+          "description": "",
+          "$ref": "#/definitions/Capabilities"
+        },
+        "callbackUrl": {
```

Comment:
Optional: Thoughts of grouping callback URL and application id together inside a model class?

Reply:
I had talked to Sabya. callbackUrl and applicationId are ortogonal concepts and don't need to be set together. So I wouldn't want to group them.